src/veins/base/phyLayer/BasePhyLayer.cc

```diff
@@ -553,29 +534,45 @@ void BasePhyLayer::handleUpperMessage(cMessage* msg)
     sendMessageDown(frame);
 }
 
+AirFrame* BasePhyLayer::createAirFrame(cPacket* macPkt)
+{
+    return new AirFrame(macPkt->getName(), AIR_FRAME);
+}
+
+void BasePhyLayer::attachSignal(AirFrame* airFrame, MacToPhyControlInfo* ctrlInfo)
+{
+    // FIXME: remove once templated signal is fully working
+    Signal signal(ctrlInfo->spectrum, ctrlInfo->startTime, ctrlInfo->duration);
+    assert(signal.getDuration() > 0);
+    auto freqIndex = ctrlInfo->spectrum.indexOf(ctrlInfo->centerFrequency);
+    signal.at(freqIndex - 1) = ctrlInfo->txPower_mW;
```

Comment:
This might be a bit too specific for BasePhyLayer. Should likely be moved to 802.11p-specific PHY